File: src/main/java/net/theevilreaper/vulpes/api/model/ItemModel.java
package net.theevilreaper.vulpes.api.model;

import io.micronaut.data.annotation.Id;
import io.micronaut.data.annotation.MappedEntity;
import io.micronaut.serde.annotation.Serdeable;

import java.util.List;
import java.util.Map;

@Serdeable
@MappedEntity("items")
public record ItemModel(
        @Id String id,
        String modelName,
        String name,
        String description,
        String displayName,
        String material,
        String group,
        int customModelData,
        int amount,
        Map<String, Short> enchantments,
        List<String> lore,
        List<String> flags
) implements VulpesModel {

}

File: src/main/kotlin/net/theevilreaper/vulpes/api/model/ItemModel.kt
package net.theevilreaper.vulpes.api.model

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document

/**
 * @author theEvilReaper
 * @version 1.0.0
 * @since
 **/
@Document("items")
data class ItemModel(
    @Id val id: String?,
    val modelName: String,
    val name: String?,
    val description: String?,
    val displayName: String?,
    val material: String?,
    val group: String?,
    val generatorKey: String = "ItemGenerator",
    val customModelId: Int?,
    val amount: Int?,
    val enchantments: Map<String, Short>?,
    val lore: List<String>?,
    val flags: List<String>?
) : VulpesModel
